__tests__/app.test.js

```diff
@@ -189,69 +188,196 @@ describe("GET /api/articles/:articleid/comments", () => {
+          descending: true,
+        });
       });
   });
 
-  test("should return  400 badRequest if provided id is invalid", () => {
-    const articleId = "invalid";
+  test("should return  400 Not Found if provided id is invalid", () => {
     return request(app)
-      .get(`/api/articles/${articleId}/comments`)
+      .get(`/api/articles/notanumber/comments`)
       .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad Request: ID provided is not valid");
+      });
+  });
+  test("should return  404 badRequest if provided id is non existent", () => {
+
+    return request(app)
+      .get(`/api/articles/9999/comments`)
+      .expect(404)
       .then((response) => {
         expect(response.body.msg).toBe(
-          'Bad Request: ID provided is not valid'
+          "Bad Request: ID provided has not been found."
         );
       });
   });
-  test("should return  404 badRequest if provided id is non existent", () => {
+  test("should return enpty array if the id provided  does not havea any comments", () => {
+    return request(app)
+      .get(`/api/articles/2/comments`)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.comments).toEqual(
+          []
+        );
+      });
+  });
+
+});
+
+describe("POST /api/articles/:articleid/comments", () => {
+  test("should return 200", () => {
+    const payload = {
+      username: "lurker",
+      body: "this is a comment",
+    };
+    return request(app).post("/api/articles/1/comments")
+      .send(payload)
+      .expect(200);
+  });
+  test("should update the comments table with the newest comment if all input ar valid.", () => {
+    const articleId = 1;
+    const count =
+      commentData.reduce((counter, obj) => {
+        if (obj.article_id === articleId) counter += 1;
+        return counter;
+      }, 0) + 1;
+    const payload = {
+      username: "lurker",
+      body: "this is a comment",
+    };
+    return request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .send(payload)
+      .then(({ body }) => {
+        return request(app)
+        .get(`/api/articles/${articleId}/comments`)
+        .expect(200)
+        .then(({body})=>{
+          return expect(body.comments.length).toEqual(count);
+        })
+
+      });
+  });
+  test("should return an array of object for the newest comment inserted associated with the article, with the following keys: title, article_idtopic, author,created_at,votes,image_img_url,comment_count  ", () => {
+    const articleId = 1;
+    const payload = {
+      username: "lurker",
+      body: "this is a comment",
+    };
+    return request(app)
+      .post(`/api/articles/${articleId}/comments`)
+      .send(payload)
+      .expect(200)
+      .then((response) => {
+        expect(response.body.newComment).toEqual(
+          expect.arrayContaining([
+            expect.objectContaining({
+              comment_id: expect.any(Number),
```

Comment:
this test is great - we know the username, body and article id so can pop that in